### ISSUE TITLE
App Freezes After Called markRead method

### ISSUE THREAD

**Issue by iRILLLL** [CleverTap/clevertap-ios-sdk#267]
My app will check when user select any message if they have already read or not.
If not, then call `markRead`

here's my code
```
if !message.isRead {
    CleverTap.sharedInstance()?.markRead(message)
}
```

but calling that method, my app becomes frozen. I can't do anything, I can't even click back button on navigation bar.
If I don't call that method, it works fine.

Happens on iOS 16.x, but I haven't check previous iOS version.
CleverTap SDK [5.0.1](https://github.com/CleverTap/clevertap-ios-sdk/releases/tag/5.0.1)

Any advice how to solve this issue? Is there's something wrong about my code when calling markRead method?

**Comment by akashvercetti** [comment]
@iRILLLL Could you please share your inbox related code? Also, Could you please share your stack trace from Xcode when the issue occcurs?

**Comment by akashvercetti** [comment]
@iRILLLL we have released a potential fix for this in our latest release v5.1.0. Could you please try updating to it? 
https://github.com/CleverTap/clevertap-ios-sdk/releases/tag/5.1.0

**Comment by iRILLLL** [comment]
hi @akashvercetti sorry forgot to reply, because it was urgent. my solution last time was to downgrade the sdk version to make it worked.

I will try to update to https://github.com/CleverTap/clevertap-ios-sdk/releases/tag/5.1.0

thank you very much
